Drivers/Src/CanBus.c

```diff
@@ -0,0 +1,240 @@
+ */
+void CanBusRxHandler(Can bus) {
+    OS_ERR err = 0;
+    OSSemPost(&(can_bus_receive_sem4[bus]), OS_OPT_POST_1,
+              &err);  // increment our queue counter
+    ASSERT_OS_ERROR(err);
+}
+
+/**
+ * @brief this function will be passed down to the BSP layer to trigger on
+ * TXend. Releases hold of the mailbox semaphore (Increments it to show mailbox
+ * available). Do not access directly outside this driver.
+ * @param bus The CAN bus to operate on. Should be CARCAN or MOTORCAN.
+ */
+void CanBusTxHandler(Can bus) {
+    OS_ERR err = 0;
+    OSSemPost(&(can_mail_sem4[bus]), OS_OPT_POST_1, &err);
+    ASSERT_OS_ERROR(err);
+}
+
+// wrapper functions for the interrupt customized for each bus
+void CanBusTxHandler1() { CanBusTxHandler(kCan1); }
+
+void CanBusRxHandler1() { CanBusRxHandler(kCan1); }
+void CanBusTxHandler3() { CanBusTxHandler(kCan3); }
+void CanBusRxHandler3() { CanBusRxHandler(kCan3); }
+
+/**
+ * @brief Checks each CAN ID. If an ID is not in kCanLut, set that ID to NULL
+ * @param wlist The whitelist containing the IDs to be checked
+ * @param size The size of the whitelist of IDs
+ * @return Returns the whitelist to be
+ */
+static CanId* whitelistValidator(CanId* wlist, uint8_t size) {
+    for (int i = 0; i < size; i++) {
+        CanId curr = wlist[i];
+        if (curr >= kMaxCanId || kCanLut[curr].size == 0) {
+            wlist[i] = 0;
+        }
+    }
+    return wlist;
+}
+
+ErrorStatus CanBusInit(Can bus, CanId* id_whitelist,
+                       uint8_t id_whitelist_size) {
+    // initialize CAN mailbox semaphore to 3 for the 3 CAN mailboxes that we
+    // have initialize tx
+    OS_ERR err = 0;
+    id_whitelist = whitelistValidator(id_whitelist, id_whitelist_size);
+    if (bus == kCan1) {
+        BspCanInit(bus, &CanBusRxHandler1, &CanBusTxHandler1,
+                   (uint16_t*)id_whitelist, id_whitelist_size);
+    } else if (bus == kCan3) {
+        BspCanInit(bus, &CanBusRxHandler3, &CanBusTxHandler3,
+                   (uint16_t*)id_whitelist, id_whitelist_size);
+    } else {
+        return ERROR;
+    }
+
+    OSMutexCreate(&(can_bus_tx_mutex[bus]),
+                  (bus == kCan1 ? "CAN TX Lock 1" : "CAN TX Lock 3"), &err);
+    ASSERT_OS_ERROR(err);
+
+    OSMutexCreate(&(can_bus_rx_mutex[bus]),
+                  (bus == kCan1 ? "CAN RX Lock 1" : "CAN RX Lock 3"), &err);
+    ASSERT_OS_ERROR(err);
+
+    OSSemCreate(
+        &(can_mail_sem4[bus]),
+        (bus == kCan1 ? "CAN Mailbox Semaphore 1" : "CAN Mailbox Semaphore 3"),
+        BSP_CAN_NUM_MAILBOX, &err);  // there's 3 hardware mailboxes on the
+                                     // board, so 3 software mailboxes
+    ASSERT_OS_ERROR(err);
+
+    OSSemCreate(&(can_bus_receive_sem4[bus]),
+                (bus == kCan1 ? "CAN Received Msg Queue Ctr 1"
+                              : "CAN Received Msg Queue Ctr 3"),
+                0, &err);  // create a mailbox counter to hold the messages in
```

Comment:
Does this need to be no linted? It's pretty readable anyways

Reply:
This is fine I think because the conditional is being another layer so the indent is there